internal/worker/events.go

```diff
@@ -0,0 +1,172 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.sia.tech/renterd/api"
+	"go.sia.tech/renterd/webhooks"
+	"go.uber.org/zap"
+)
+
+type (
+	EventManager interface {
```

Comment:
I'd call this `EventSubscriber` since it's the thing doing the subscribing and call the `EventSubscriber` `EventHandler` instead.